File: Examples/CSharp/Articles/GetWarningsForFontSubstitution.cs
using System;
using System.IO;
using Aspose.Cells;
using System.Diagnostics;

namespace Aspose.Cells.Examples.CSharp.Articles
{
    public class GetWarningsForFontSubstitution : IWarningCallback
    {
        public void Warning(WarningInfo info)
        {
            if (info.WarningType == WarningType.FontSubstitution)
            {
                Console.WriteLine("WARNING INFO: " + info.Description);
                Debug.WriteLine("WARNING INFO: " + info.Description);
            }
        }

        public static void Run()
        {
            //Source directory
            string sourceDir = RunExamples.Get_SourceDirectory();

            //Output directory
            string outputDir = RunExamples.Get_OutputDirectory();

            Workbook workbook = new Workbook(sourceDir + "sampleGetWarningsForFontSubstitution.xlsx");

            PdfSaveOptions options = new PdfSaveOptions();
            options.WarningCallback = new GetWarningsForFontSubstitution();

            workbook.Save(outputDir + "outputGetWarningsForFontSubstitution.pdf", options);

            Console.WriteLine("GetWarningsForFontSubstitution executed successfully.");
        }
    }    
}



            
      

File: Examples/VisualBasic/Articles/GetWarningsForFontSubstitution.vb
Namespace Articles
        ' ExStart:1
    Public Class GetWarningsForFontSubstitution
        Implements IWarningCallback
        Public Sub Warning(info As WarningInfo) Implements IWarningCallback.Warning
            If info.WarningType = WarningType.FontSubstitution Then
                Debug.WriteLine("WARNING INFO: " + info.Description)
            End If
        End Sub

        Public Shared Sub Run()
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)
            Dim workbook As New Workbook(dataDir & Convert.ToString("source.xlsx"))

            Dim options As New PdfSaveOptions()
            options.WarningCallback = New GetWarningsForFontSubstitution()
            dataDir = dataDir & Convert.ToString("output_out_.pdf")
            workbook.Save(dataDir, options)
        End Sub
    End Class
    ' ExEnd:1
End Namespace